[CI] MultiClusterSpecIT class failing

**Build Scans:**
- [elasticsearch-intake #28253 / 9.1.4_bwc-snapshots](https://gradle-enterprise.elastic.co/s/qruczdhjavibo)
- [elasticsearch-pull-request #91240 / 9.2.0_part_3_bwc-snapshots](https://gradle-enterprise.elastic.co/s/uxbj2urcxp4v6)
- [elasticsearch-periodic #10557 / openjdk23_8.19.4_java-matrix-bwc](https://gradle-enterprise.elastic.co/s/ann5scn3on3x6)
- [elasticsearch-periodic #10557 / 8.19.4_bwc](https://gradle-enterprise.elastic.co/s/gakzayxeryhrg)
- [elasticsearch-pull-request #91242 / 8.19.4_part_3_bwc-snapshots](https://gradle-enterprise.elastic.co/s/mdpohmoyoatkw)
- [elasticsearch-pull-request #91242 / 9.0.7_part_3_bwc-snapshots](https://gradle-enterprise.elastic.co/s/uktt3qvsg7wxo)
- [elasticsearch-pull-request #91242 / 9.1.4_part_3_bwc-snapshots](https://gradle-enterprise.elastic.co/s/ednh6wg2l4mye)
- [elasticsearch-pull-request #91234 / 9.0.7_part_3_bwc-snapshots](https://gradle-enterprise.elastic.co/s/adccms4q6jpg6)
- [elasticsearch-pull-request #91242 / 9.2.0_part_3_bwc-snapshots](https://gradle-enterprise.elastic.co/s/2dj3l2ugr26dk)
- [elasticsearch-pull-request #91217 / 9.0.7_part_3_bwc-snapshots](https://gradle-enterprise.elastic.co/s/kc36etpvpwc3g)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:multi-clusters:v9.1.4#newToOld" -Dtests.class="org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT" -Dtests.method="test {csv-spec:spatial.CentroidFromAirportsAfterIntersectsEvalExpression}" -Dtests.seed=A0439B8C19C9F735 -Dtests.bwc=true -Dtests.locale=ms-SG -Dtests.timezone=America/Anchorage -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: java.util.concurrent.ExecutionException: java.lang.RuntimeException: Failed to create indices in parallel
```

**Issue Reasons:**
- [main] 32 failures in class org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT (3.2% fail rate in 1000 executions)
- [main] 5 failures in step 9.2.0_part_3_bwc-snapshots (7.1% fail rate in 70 executions)
- [main] 9 failures in step 9.0.7_part_3_bwc-snapshots (12.7% fail rate in 71 executions)
- [main] 7 failures in step 9.1.4_part_3_bwc-snapshots (9.7% fail rate in 72 executions)
- [main] 7 failures in step 8.19.4_part_3_bwc-snapshots (9.6% fail rate in 73 executions)
- [main] 2 failures in pipeline elasticsearch-intake (12.5% fail rate in 16 executions)
- [main] 18 failures in pipeline elasticsearch-pull-request (24.3% fail rate in 74 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/CsvTestsDataLoader.java
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/CsvTestsDataLoader.java
@@ -17,7 +17,6 @@ import org.apache.http.auth.UsernamePasswordCredentials;
 import org.apache.http.client.CredentialsProvider;
 import org.apache.http.impl.client.BasicCredentialsProvider;
 import org.apache.logging.log4j.core.config.plugins.util.PluginManager;
-import org.apache.lucene.util.IOConsumer;
 import org.elasticsearch.client.Request;
 import org.elasticsearch.client.Response;
 import org.elasticsearch.client.ResponseException;
@@ -31,7 +30,6 @@ import org.elasticsearch.core.Nullable;
 import org.elasticsearch.inference.TaskType;
 import org.elasticsearch.logging.LogManager;
 import org.elasticsearch.logging.Logger;
-import org.elasticsearch.test.ESTestCase;
 import org.elasticsearch.test.rest.ESRestTestCase;
 import org.elasticsearch.xcontent.XContentType;
 
@@ -45,7 +43,6 @@ import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
-import java.util.concurrent.Semaphore;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
@@ -58,7 +55,6 @@ import static org.elasticsearch.xpack.esql.CsvTestUtils.multiValuesAwareCsvToStr
 import static org.elasticsearch.xpack.esql.EsqlTestUtils.reader;
 
 public class CsvTestsDataLoader {
-    private static final int PARALLEL_THREADS = 10;
     private static final int BULK_DATA_SIZE = 100_000;
     private static final TestDataset EMPLOYEES = new TestDataset("employees", "mapping-default.json", "employees.csv").noSubfields();
     private static final TestDataset EMPLOYEES_INCOMPATIBLE = new TestDataset(
@@ -404,35 +400,16 @@ public class CsvTestsDataLoader {
         IndexCreator indexCreator
     ) throws IOException {
         Logger logger = LogManager.getLogger(CsvTestsDataLoader.class);
-        List<TestDataset> datasets = availableDatasetsForEs(supportsIndexModeLookup, supportsSourceFieldMapping, inferenceEnabled).stream()
-            .toList();
 
-        executeInParallel(datasets, dataset -> createIndex(client, dataset, indexCreator), "Failed to create indices in parallel");
-
-        executeInParallel(datasets, dataset -> loadData(client, dataset, logger), "Failed to load data in parallel");
-
-        forceMerge(client, datasets.stream().map(d -> d.indexName).collect(Collectors.toSet()), logger);
-
-        executeInParallel(
-            ENRICH_POLICIES,
-            policy -> loadEnrichPolicy(client, policy.policyName, policy.policyFileName, logger),
-            "Failed to load enrich policies in parallel"
-        );
-
-    }
-
-    private static <T> void executeInParallel(List<T> items, IOConsumer<T> consumer, String errorMessage) {
-        Semaphore semaphore = new Semaphore(PARALLEL_THREADS);
-        ESTestCase.runInParallel(items.size(), i -> {
-            try {
-                semaphore.acquire();
-                consumer.accept(items.get(i));
-            } catch (IOException | InterruptedException e) {
-                throw new RuntimeException(errorMessage, e);
-            } finally {
-                semaphore.release();
-            }
-        });
+        Set<String> loadedDatasets = new HashSet<>();
+        for (var dataset : availableDatasetsForEs(supportsIndexModeLookup, supportsSourceFieldMapping, inferenceEnabled)) {
+            load(client, dataset, logger, indexCreator);
+            loadedDatasets.add(dataset.indexName);
+        }
+        forceMerge(client, loadedDatasets, logger);
+        for (var policy : ENRICH_POLICIES) {
+            loadEnrichPolicy(client, policy.policyName, policy.policyFileName, logger);
+        }
     }
 
     public static void createInferenceEndpoints(RestClient client) throws IOException {
@@ -589,13 +566,11 @@ public class CsvTestsDataLoader {
         return result;
     }
 
-    private static void createIndex(RestClient client, TestDataset dataset, IndexCreator indexCreator) throws IOException {
+    private static void load(RestClient client, TestDataset dataset, Logger logger, IndexCreator indexCreator) throws IOException {
         URL mapping = getResource("/" + dataset.mappingFileName);
         Settings indexSettings = dataset.readSettingsFile();
         indexCreator.createIndex(client, dataset.indexName, readMappingFile(mapping, dataset.typeMapping), indexSettings);
-    }
 
-    private static void loadData(RestClient client, TestDataset dataset, Logger logger) throws IOException {
         // Some examples only test that the query and mappings are valid, and don't need example data. Use .noData() for those
         if (dataset.dataFileName != null) {
             URL data = getResource("/data/" + dataset.dataFileName);

--- a/x-pack/plugin/esql/qa/testFixtures/src/test/java/org/elasticsearch/xpack/esql/CsvTestsDataLoaderTests.java
+++ b/x-pack/plugin/esql/qa/testFixtures/src/test/java/org/elasticsearch/xpack/esql/CsvTestsDataLoaderTests.java
@@ -11,18 +11,12 @@ import org.elasticsearch.test.ESTestCase;
 
 import java.net.ConnectException;
 
-import static org.hamcrest.Matchers.instanceOf;
 import static org.hamcrest.Matchers.startsWith;
 
 public class CsvTestsDataLoaderTests extends ESTestCase {
 
     public void testCsvTestsDataLoaderExecution() {
-        Throwable cause = expectThrows(AssertionError.class, () -> CsvTestsDataLoader.main(new String[] {}));
-        // find the root cause
-        while (cause.getCause() != null) {
-            cause = cause.getCause();
-        }
-        assertThat(cause, instanceOf(ConnectException.class));
-        assertThat(cause.getMessage(), startsWith("Connection refused"));
+        ConnectException ce = expectThrows(ConnectException.class, () -> CsvTestsDataLoader.main(new String[] {}));
+        assertThat(ce.getMessage(), startsWith("Connection refused"));
     }
 }

